src/gui/rss/rsswidget.cpp

```diff
@@ -375,6 +382,89 @@ void RSSWidget::downloadSelectedTorrents()
     }
 }
 
+// Provides a way to filter a torrent's name. Used for Right-Click -> Add new RSS Rule.
+const QString RSSWidget::sanitizeNewRssRuleName(const QString &ruleName)
+{
+    QString name;
+
+    for (QChar currentChar : ruleName)
+    {
+        if (currentChar.isSpace() || currentChar.isLetterOrNumber())
+        {
+            name.append(currentChar);
+        }
+    }
+
+    return name.simplified(); // return without multiple whitespace
+}
+
+void RSSWidget::newRSSRuleBtn() {
+
+    // Safety:
+    // The feature only appears if there is only 1 article selected,
+    // but just to make sure, we do another check here.
+    if (m_articleListWidget->selectedItems().count() != 1)
+        return;
+
+    // Retrieve article from selected items
+    QListWidgetItem *selectedRssRule = asConst(m_articleListWidget->selectedItems()[0]);
+    RSS::Article *selectedRssRuleArticle = selectedRssRule->data(Qt::UserRole).value<RSS::Article *>();
```

Comment:
It's better to avoid such "too much distinctive" names if there are no other ones to distinguish from. Just `article` would be enough if there are no other articles in current context.